client/actions/assignments/api.ts

```diff
@@ -605,21 +541,21 @@ const removeAssignment = (assignment) => (
     )
 );
 
-const unlink = (assignment) => (
-    (dispatch, getState, {api, notify}) => (
+function unlink(assignment: IAssignmentItem) {
+    return (dispatch, getState, {api, notify}) => (
         api('assignments_unlink').save({}, {
             assignment_id: assignment._id,
             item_id: get(assignment, 'item_ids[0]'),
         })
             .then(() => {
                 notify.success(gettext('Assignment reverted.'));
-                return dispatch(self.unlock(assignment));
+                return planningApi.locks.unlockItem(assignment);
```

Comment:
I'm not sure about those changes in calling the actions in general, what's the motivation for that?

Reply:
We originally used Redux not just for state management, but as a way to pass AngularJS services from client-core as well. Sometimes we were dispatching an action that never, directly, updated the Redux state. This added an extra layer that didn't add any benefits.

I would rather see Redux being used for state management only, and use the new pure-js API provided by client-core to access functionality provided by client-core. This would mean, eventually, this `actions` folder would only contain dispatch actions that update state and no other side affects.

Therefor actions themselves can be implemented in this internal Planning API. This internal API can be executed from anywhere without acquiring the Redux store, and if need be this API will dispatch the appropriate actions to update the store.

Reply:
I'm just not sure if implicit access to dispatch in this internal api makes things simpler or more complex at the end if it can be called from anywhere anytime. not against it per se rather wondering if there can be any side effects.